src/robot/ros/RosRobot.cpp

```diff
@@ -38,6 +52,262 @@ RosRobot::RosRobot(
+  for (auto bodyNode : refSkeleton->getBodyNodes())
+  {
+    if (!mMetaSkeleton->hasBodyNode(bodyNode))
+    {
+      dtwarn << "Subrobot '" << name << "'' contains body node "
+             << bodyNode->getName() << " not in parent MetaSkeleton."
+             << std::endl;
+      return nullptr;
+    }
+  }
+
+  // Ensure subrobot DoFs are disjoint
+  for (const auto& subrobot : mSubRobots)
+  {
+    auto dofs = subrobot.second->getDofs();
+    for (std::string dofName : util::dofNamesFromSkeleton(refSkeleton))
+    {
+      if (dofs.find(dofName) != dofs.end())
+      {
+        dtwarn << "Subrobot '" << name << "'' overlaps existing subrobot "
+               << subrobot.first << " at DoF " << dofName << "." << std::endl;
+        return nullptr;
+      }
+    }
+  }
+
+  // Create the sub RosRobot.
+  auto subRosRobot = std::make_shared<RosRobot>(
+      refSkeleton, this, mCollisionDetector, mSelfCollisionFilter, name);
+  mSubRobots[name] = subRosRobot;
+  return subRosRobot;
+}
+
+//=============================================================================
+void RosRobot::deactivateExecutor()
+{
+  if (!mActiveExecutor.empty())
+  {
+    if (!mRosControllerNames[mActiveExecutor].empty()
+        && !stopController(mRosControllerNames[mActiveExecutor]))
+    {
+      throw std::runtime_error(
+          "Could not stop controller: " + mRosControllerNames[mActiveExecutor]);
+    }
+    Robot::deactivateExecutor();
+  }
+}
+
+//=============================================================================
+std::string RosRobot::registerExecutor(
+    aikido::control::ExecutorPtr executor,
+    std::string controllerName,
+    hardware_interface::JointCommandModes controllerMode,
+    std::string desiredName)
+{
+  if (!controllerName.empty() && !loadController(controllerName))
+  {
+    throw std::runtime_error("Could not load controller: " + controllerName);
+  }
+  std::string id = Robot::registerExecutor(executor,desiredName);
+  if (!id.empty())
+  {
+    mRosControllerNames[id] = controllerName;
+    mRosControllerModes[id] = controllerMode;
+  }
+  return id;
+}
+
+std::string RosRobot::registerExecutor(
+    aikido::control::ExecutorPtr executor, 
+    std::string controllerName,
+    std::string desiredName)
+{
+  return registerExecutor(
+      executor, controllerName, hardware_interface::JointCommandModes::ERROR, desiredName);
+}
+
+//=============================================================================
+std::string RosRobot::registerExecutor(aikido::control::ExecutorPtr executor, std::string desiredName)
+{
+  return registerExecutor(
+      executor, std::string(), hardware_interface::JointCommandModes::ERROR, desiredName);
+}
+
+//=============================================================================
+bool RosRobot::activateExecutor(std::string id)
+{
+  // Deactivate active executor
+  deactivateExecutor();
+
+  // Validate input
+  if (mExecutors.find(id) == mExecutors.end())
+  {
+    return false;
```

Comment:
add a warning here (not sure if Robot does this, but would be a good idea there as well)